backend/src/jv2backend/journalClasses.py

```diff
@@ -0,0 +1,310 @@
+        start, end = _to_datetime(start.strip()), _to_datetime(end.strip())
+    except ValueError:  # bad format
+        return pd.DataFrame()
+
+    column_values = pd.to_datetime(data[column])
+    return data[column_values.between(start, end, inclusive="both")]
+
+
+# Map a field name to a handler for that query if it should have special
+# handling
+_SPECIAL_QUERY_HANDLERS = {
+    "experiment_identifier": equals,
+    "run_number": inrange_int,
+    "start_time": inrange_datetime,
+}
+
+
+# JournalData class
+class JournalData:
+    """JournalData captures all run information from a given journal file"""
+
+    def __init__(self, source: str, data: pd.DataFrame) -> None:
+        """
+        :param source: Defines the source filename
+        """
+        self._source = source
+        self._data = data
+
+    @property
+    def instrument(self) -> str:
+        return self._source
+
+    @property
+    def run_count(self) -> int:
+        """Return the number of runs listed within this Journal"""
+        return len(self._data)
+
+    @property
+    def get_last_run_number(self) -> int:
+        """Return the run number of the last run in the journal"""
+        run_numbers = self._data["run_number"]
+        return int(run_numbers.max())
+
+    def run(self, run_number: str) -> Optional[dict]:
+        """Return a dictionary describing the given run number
+
+        :param run_number: Run number to select
+        :return: A dict describing the Run or None if the run does not exist
+        """
+        matches = self._data[self._data["run_number"] == run_number]
+        if len(matches) == 0:
+            return None
+        else:
+            return matches.to_dict(orient="records")[0]
+
+    def search(
+        self, run_field: str, user_input: str, case_sensitive: bool = False
+    ) -> JournalData:
+        """Search data for runs whose run_field matches the user_input"""
+        # Different fields need handling differently but we will fallback to a
+        # basic "is in string check"
+        query_handle = _SPECIAL_QUERY_HANDLERS.get(run_field, contains)
+
+        return JournalData(
+            self.source,
+            query_handle(self._data, run_field, user_input, case_sensitive),
+        )
+
+    # Output formats
+    def to_json(self) -> str:
+        return self._data.to_json(orient="records")
+
+
+# Operations on multiple journal data
+def concatenate(journals: Sequence[JournalData]) -> JournalData:
+    """Concatenate the Journals to a single Journal
+
+    :param journals: Sequence of Journal objects
+    :return: A new JournalData, the result of concatenating the input data
+    """
+    return JournalData(
+        journals[0].source, pd.concat([journal._data for journal in journals])
+    )
+
+
+@dataclass
+class BasicJournalFile:
+    """Defines basic properties of a single journal file"""
+    server_root: str
+    directory: str
+    filename: str
+    data_directory: str = "/I/AM/A/TEST/PATH"
```

Comment:
```suggestion
    filename: str    
    _ : KW_ONLY
    data_directory: str = "/I/AM/A/TEST/PATH"
```

Having default attributes in dataclasses that will be inherrited might cause problems, as python does not allow positional attributes to come after defaults (which may happen when this class is inherrited). It's better for readability and future compatibility to specify these as keyword-only (every attribute after the `_: KW_ONLY` will be keyword-only). Import `KW_ONLY` from `dataclasses`.